packages/e2e-tests/utils/generateNewPatient.ts

```diff
@@ -0,0 +1,90 @@
+import { faker } from '@faker-js/faker';
+import { AllPatientsPage } from '../pages/patients/AllPatientsPage';
+import { constructFacilityUrl } from './navigation';
+
+function generateNHN() {
+  const letters = faker.string.alpha({ length: 4, casing: 'upper' });
+  const numbers = faker.string.numeric(6);
+  const generatedId = `${letters}${numbers}`;
+
+  return generatedId;
+}
+
+function generatePatientData() {
+  const gender = faker.helpers.arrayElement(['male', 'female']);
+  const firstName = faker.person.firstName(gender);
+  const lastName = faker.person.lastName();
+  const dob = faker.date.birthdate({ min: 18, max: 80, mode: 'age' });
+  const formattedDOB = dob.toISOString().split('T')[0]; // Convert to YYYY-MM-DD format
+  const nhn = generateNHN();
+
+  return { firstName, lastName, gender, formattedDOB, nhn };
+}
+
```

Comment:
not entirely sure in what context this package exists but we have the `fake-data` package which can much more concisely generate patients:

```js
import { fake } from '@tamanu/fake-data/fake';
const fakePatientData = fake(Patient);
```

and there's the `api-client` package as well which should help with the API a bit.

Reply:
Thanks for this suggestion! Flynn and I both attempted to get `fake-data` to work in this package but it seems relatively complex and probably out of the scope of this PR. The current implementation seems sufficient for now and we can invest more time into getting `fake-data` to work later if needed 🙏